### DISCUSSION TITLE
Please add a datetime stamp at the beginning of the stdout messages.

### DISCUSSION THREAD

**Discussion by RimBlock** [madMAx43v3r/chia-plotter#849] [Ideas]
Hi,

I tend to run the plotter on Ubuntu using nohup so I do nto need to be logged in to keep the process going. 

I can view the job output from tailing the nohup.out file when I want to check on progress.

I have a few plotters and so am integrating them with a log aggregator.  It would be very helpful if the plotter user messages were prefixed with the datetime stamp enabling accurate recording of when that event happened.

Without it the log aggregater will make a best guess based on when it was checked and found.  It would also make it easy to tell if the process has stalled without watching for the next log message.

Thanks

**Comment by SebMoore** [comment]
> Hi,
> 
> I tend to run the plotter on Ubuntu using nohup so I do nto need to be logged in to keep the process going.
> 
> I can view the job output from tailing the nohup.out file when I want to check on progress.
> 
> I have a few plotters and so am integrating them with a log aggregator. It would be very helpful if the plotter user messages were prefixed with the datetime stamp enabling accurate recording of when that event happened.
> 
> Without it the log aggregater will make a best guess based on when it was checked and found. It would also make it easy to tell if the process has stalled without watching for the next log message.
> 
> Thanks


Obligatory "why don't you use tmux 🙄" comment here.
Not that it solves your problem, I just prefer it.
Anyway, actual answer: you can use `ts` from the package `moreutils`: this will essentially print a date/timestamp to every line of terminal output.
`sudo apt-get install -y moreutils`
`./chia-plotter blah blah options | ts`
manpage is [here](https://manpages.debian.org/testing/moreutils/ts.1.en.html), but it works pretty well stock.

**Reply by RimBlock** [reply]
Thanks, very helpful.  Will take a look at both.

I tend to be pretty oldskool having started working on Linux 30+ years ago so still use vi etc.  A little mordernisation maybe :-).

Would be good to still have it added in source rather than applying external fixes.  

Unfortunately looking at the code, stdout seems to be hard coded for each message rather than using a single helper logging function and sending messages to that for it to then output.  Advantage being that adding a timestamp in one place would have been easy.  Custom logging would also make ir pretty easy incorporate (i.e. to files and possibly rsyslog).

Whilst I do program in some languages, cpp is not one I  am perficient in  :-(.  Would have been happy to have fixed and put a pull request in for it.
